packages/pnpm/src/main.ts

```diff
@@ -163,7 +163,16 @@ export default async function run (inputArgv: string[]) {
       }
       process.exit(0)
     }
-    const filterResults = await filterPackages(allProjects, config.filter ?? [], {
+
+    config.filter = config.filter ?? []
+    config.filterProd = config.filter ?? []
+
+    const filters = [
+      ...config.filter.reduce<FilterObj[]>((filters, filter) => ([...filters, { filter, followProdDepsOnly: false }]), []),
+      ...config.filterProd.reduce<FilterObj[]>((filters, filter) => ([...filters, { filter, followProdDepsOnly: true }]), []),
```

Comment:
a simple map will work just well

```suggestion
      ...config.filterProd.map<FilterObj[]>((filter) => ({ filter, followProdDepsOnly: true })),
```